singer_sdk/streams/core.py

```diff
@@ -970,9 +969,45 @@ def finalize_state_progress_markers(self, state: Optional[dict] = None) -> None:
 
     # Private sync methods:
 
-    def _sync_records(  # noqa C901  # too complex
-        self, context: Optional[dict] = None
+    def _process_record(
+        self,
+        record: dict,
+        selected: bool,
+        record_context: Optional[dict] = None,
+        child_context: Optional[dict] = None,
+        partition_context: Optional[dict] = None,
+        count: int = 0,
     ) -> None:
+        """Process a record.
+
+        Args:
+            record: The record to process.
+            selected: Whether the stream is selected.
+            record_context: The record context.
+            child_context: The child context.
+            partition_context: The partition context.
+            count: The current record count.
```

Comment:
Would be good in the docstring to clarify - is `count` per batch, per stream, per the above contexts, or total?

Reply:
I think 'per stream and (if applicable) per parent context or partition' - but I could be wrong on that.

Reply:
Updated!

Reply:
👍 🙏